From 12d6b0fe6251e77dbda8c3daeefacb028a267750 Mon Sep 17 00:00:00 2001
From: Benjamin Otte <otte@redhat.com>
Date: Thu, 05 Apr 2012 09:30:41 +0000
Subject: paned: Shrinkable widgets don't require a size

This could otherwise lead to very funny size request behaviors, in
particular in empathy.

https://bugzilla.gnome.org/show_bug.cgi?id=587441
---
diff --git a/gtk/gtkpaned.c b/gtk/gtkpaned.c
index f2fbe13..5b374b2 100644
--- a/gtk/gtkpaned.c
+++ b/gtk/gtkpaned.c
@@ -904,7 +904,10 @@ gtk_paned_get_preferred_size (GtkWidget      *widget,
   if (priv->child1 && gtk_widget_get_visible (priv->child1))
     {
       get_preferred_size_for_size (priv->child1, orientation, size, &child_min, &child_nat);
-      *minimum = child_min;
+      if (priv->child1_shrink && priv->orientation == orientation)
+        *minimum = 0;
+      else
+        *minimum = child_min;
       *natural = child_nat;
     }
 
@@ -914,7 +917,8 @@ gtk_paned_get_preferred_size (GtkWidget      *widget,
 
       if (priv->orientation == orientation)
         {
-          *minimum += child_min;
+          if (!priv->child2_shrink)
+            *minimum += child_min;
           *natural += child_nat;
         }
       else
--
cgit v0.9.0.2

